pkg/kubelet/kubelet_getters.go

```diff
@@ -179,7 +179,9 @@ func (kl *Kubelet) GetPods() []*v1.Pod {
 	for _, p := range pods {
 		if kubelettypes.IsStaticPod(p) {
 			if status, ok := kl.statusManager.GetPodStatus(p.UID); ok {
-				klog.V(2).InfoS("Pod status updated", "pod", klog.KObj(p), "status", status.Phase)
+				var statusOnlyPhase = v1.PodStatus{}
+				statusOnlyPhase.Phase = status.Phase
+				klog.V(2).InfoS("Pod status updated", "pod", klog.KObj(p), "status", statusOnlyPhase)
```

Comment:
```
klog.V(2).InfoS("Pod status updated", "pod", klog.KObj(p), "phase", status.Phase)
```
Is this ok?

https://github.com/kubernetes/kubernetes/blob/442574f3a7321ff0f34dec9bae67f28651dbabd2/pkg/kubelet/kubelet_pods.go#L1463-L1469

I found there are some similar logs here.

Reply:
I found this example [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/prober/worker.go#L215).

But i thought it's not good idea.
If we do this, the status of the pod in the logs will be in 2 places - `status.phase` and `phase`. When someone analyzes the logs, this can create difficulties.
May be I add new function or method like `GetPodPhase` which return `v1.PodStatus{}` only with phase?

Reply:
I can see the logic behind this decision.

However, is there ever a situation where "status" is logged with all fields? If not, then always logging "phase" or "status.phase" makes more sense to me. Just my two cents.